### ISSUE TITLE
[Feature]: Implement Ed25519 signature

### ISSUE THREAD

**Issue by ggonzalez94** [OpenZeppelin/rust-contracts-stylus#260]
### What is the feature you would like to see?

Reference implementation: https://github.com/RustCrypto/signatures/tree/master/ed25519

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

**Comment by programskillforverification** [comment]
I will work on this.

**Comment by bidzyyys** [comment]
Hey @programskillforverification! Thank you for your interest in contribution 🚀 
Go ahead and update us whenever you need any assistance or take a design decision.

**Comment by ggonzalez94** [comment]
@programskillforverification awesome to have you as a recurring contributor 🚀 - just make sure we prioritize #275 and get that merged ASAP.
Looking forward to seeing your PR on this one!

**Comment by programskillforverification** [comment]
@ggonzalez94 so we don't use `Signature` in reference implementation and implement an our own ed25519?

**Comment by qalisander** [comment]
@programskillforverification correct. I'll gonna add [elliptic curves primitives](https://github.com/OpenZeppelin/rust-contracts-stylus/issues/343) first and based on them add ed25519 curve. After it will be possible to have a signature based on that curve

**Comment by bidzyyys** [comment]
Blocked by #343
